lib/cocina/models/collection.rb

```diff
@@ -30,6 +30,7 @@ class Administrative < Struct
 
       # Identification sub-schema for the Collection
       class Identification < Struct
+        attribute :agreementId, Types::Strict::String.meta(omittable: true)
```

Comment:
Looks like this should be `hasAgreement` in the structural schema: https://github.com/sul-dlss/cocina-models/blob/315b55abbb3b9980013ca1419aa51a991a9973ac/docs/maps/DRO.json#L270

Reply:
This pr is specifically about the identityMetadata (as posted in the https://github.com/sul-dlss/dor-services-app/issues/683 issue). Should that be udpated?

Reply:
@jcoyne updated. 

Do collections have agreements or just items? happy to pair if this isn't what we were thinking.